borg/helpers.py

```diff
@@ -348,6 +348,33 @@ def __str__(self):
         return self.pattern_orig
 
 
+class ExcludeRegex(IncludePattern):
+    """Regular expression to exclude.
+    """
+    def __init__(self, pattern):
+        self.pattern_orig = pattern
+        self.match_count = 0
+
+        if sys.platform in ('darwin',):
+            pattern = unicodedata.normalize("NFD", pattern)
+
+        self.pattern = pattern
+        self.regex = re.compile(pattern)
+
+    @normalized
+    def match(self, path):
+        matches = self.regex.search(path) is not None
```

Comment:
re.match requires the full string to be matched. If someone wanted to exclude everything in `/tmp` they'd have to wrote `/tmp/.*`. re.search matches grep's behaviour where a partial, optionally anchored match is sufficient, i.e. `^/tmp/`.

This was a deliberate decision. My initial implementation used re.match, but I found it cumbersome to use.